nix/default.nix

```diff
@@ -233,7 +233,7 @@ let
 
   # define the base requirements
   base-req = [ pkgs.bashInteractive pkgs.coreutils pkgs.gnumake ];
-  codegen-req = [ pkgs.protobuf pkgs.ocamlPackages.ocaml-protoc pkgs.glibc ]
+  codegen-req = [ pkgs.protobuf pkgs.ocamlPackages.ocaml-protoc pkgs.glibc pkgs.oc ]
```

Comment:
Unfortunately `oc` is not available in Nix, and since we are not going to codegen the k8s resources, let's remove that.

Reply:
We are going to do that in the next milestone, but I am fine with removing it for now

Reply:
Seems kubectl is available
https://nixos.wiki/wiki/Kubernetes

So let's use that!

Reply:
I meant, this list define the tool used in the Makefile.

Reply:
Yeah, and I guess a part of the deployment in nix would be to use the kubectl?

Reply:
I think it's ok to skip those requirements in nix for now, for example we don't manage docker-compose or podman here (e.g. we assume the tools are already available to the operator).

We can add those tool in nix, but let's use another list, e.g. `deploy-req = [ pkgs.kubectl pkgs.docker-compose pkgs.podman ];`

Reply:
Sure!
I just dipped into the nix thing and am excited to learn about it

I can do that

Reply:
(ps, some people still use good old docker)

Reply:
I am not sure it is need to mix the nix update with this PR. Let's push it to the following one.

Reply:
pkgs.oc can be removed.